File: backend/ecommerce/src/main/java/com/bookstore/ecommerce/app/order/command/Customer.java
package com.bookstore.ecommerce.app.order.command;

import lombok.Data;

@Data
public class Customer {
  private String name;
  private String phone;
  private String deliveryAddress;

  public com.bookstore.ecommerce.app.domain.data.Customer toDataObject() {
    return com.bookstore.ecommerce.app.domain.data.Customer.builder()
      .name(this.getName())
      .phone(this.getPhone())
      .deliveryAddress(this.getDeliveryAddress())
      .build();
  }
}

File: order/app/order/command/customer.go
package command

import "ecommerce/app/domain/data"

type Customer struct {
	Name            string
	Phone           string
	DeliveryAddress string
}

func (c *Customer) toDataObject() data.Customer {
	return data.Customer{Name: c.Name, Phone: c.Phone, DeliveryAddress: c.DeliveryAddress}
}
